lib/models/s3-event.js

```diff
@@ -0,0 +1,79 @@
+'use strict';
+const crypto = require('crypto');
+
+const S3Event = function (eventData, reqParams) {
+  const {reqHeaders,sourceIp} = reqParams;
+  let eventName = '';
+  const s3Object = {
+    "key": eventData.S3Item.key,
+    "sequencer": ('00' + (Math.floor(Date.now())).toString(16)).toUpperCase()
+  };
+  switch (eventData.eventType) {
+    case 'Copy':
+      eventName = 'ObjectCreated:Copy';
+      s3Object.size = eventData.S3Item.size;
+      break;
+
+    case 'Put':
+      eventName = 'ObjectCreated:Put';
+      s3Object.size = eventData.S3Item.size;
+      s3Object.eTag = eventData.S3Item.md5;
+      break;
+
+    case 'Post':
+      eventName = 'ObjectCreated:Post';
+      s3Object.size = eventData.S3Item.size;
+      s3Object.eTag = eventData.S3Item.md5;
+      break;
+
+    case 'Delete':
+      eventName = 'ObjectRemoved:Delete';
+      break;
+  }
+
+  const event = {
+    "Records": [
+      {
+        "eventVersion": "2.0",
+        "eventSource": "aws:s3",
+        "awsRegion": reqHeaders.host,
+        "eventTime": eventData.S3Item.creationDate || new Date().toISOString(),
+        "eventName": eventName,
+        "userIdentity": {
+          "principalId": 'AWS:' + randomString(21).toUpperCase(),
```

Comment:
The `userIdentity.principalId` field shown in the [example](https://docs.aws.amazon.com/AmazonS3/latest/dev/notification-content-structure.html) isn't prepended with `AWS:`. Please remove the prefix.